### PR TITLE
Rectify: Contract-vs-Reality Divergence Immunity

### LINKED ISSUE #1644: review-pr contract divergence: approved_with_comments verdict causes adjudicated_failure (35/41 failures)
## Summary

The `review-pr` skill contract in `skill_contracts.yaml` diverges from the SKILL.md specification for the `approved_with_comments` verdict, causing **35 of 41 `adjudicated_failure` sessions** (85%) in the last 500 logged sessions. The skill correctly completes its review and emits the correct verdict, but the adjudication system rejects the output because the contract requires a `%%REVIEW_GATE%%` tag that the SKILL.md explicitly instructs the model not to emit for this verdict.

## Root Cause

`skill_contracts.yaml` lines 552-554 define two `expected_output_patterns` with AND semantics:

```yaml
expected_output_patterns:
- "verdict\\s*=\\s*(approved|changes_requested|needs_human)"
- "%%REVIEW_GATE::(LOOP_REQUIRED|CLEAR)%%"
```

But `review-pr/SKILL.md` line 679 explicitly instructs:
> If `verdict = approved_with_comments`: do NOT emit a gate tag

When approximately half of reviewed PRs produce warning-level findings (no criticals), the skill correctly emits `verdict = approved_with_comments` with no gate tag — exactly as instructed. `_check_expected_patterns` (`_session_content.py:53-63`) rejects the output because pattern 2 is absent (AND semantics: ALL patterns must match).

### Additional mismatches in the same contract entry

1. **`allowed_values`** (lines 548-551) lists only `approved`, `changes_requested`, `needs_human` — `approved_with_comments` is absent
2. **`pattern_examples`** (lines 556-558) contains only 3 examples, none for `approved_with_comments`
3. **Verdict pattern alternation** excludes `approved_with_comments` — though `re.search` prefix-matches `approved` inside `approved_with_comments`, this is accidental, not intentional

### Already-correct surfaces

The recipe-level contract cards (`recipes/contracts/implementation.yaml:148-154`, `remediation.yaml:176-179`, `implementation-groups.yaml:165-168`) have already been updated to include `approved_with_comments` in the verdict pattern and omit the `%%REVIEW_GATE%%` requirement. The authoritative runtime source (`skill_contracts.yaml`) was never synchronized.

## Evidence from Session Logs

- 35 of 41 `adjudicated_failure` entries are `/review-pr` sessions
- All 35 have `cli_subtype: success`, `exit_code: 0` — the CLI never reports failure
- 31 of 35 have `write_call_count: 0` → terminal failure (no CONTRACT_RECOVERY)
- 4 of 35 have `write_call_count > 0` → promoted to CONTRACT_RECOVERY (retriable) but burns retry budget
- Non-determinism confirmed: same PR, same version, same anomaly count → different outcomes depending on whether the model chose `approved` vs `approved_with_comments`
- The ~50% failure rate directly corresponds to the ~50% of PRs that receive `approved_with_comments`

### 1. `skill_contracts.yaml` — review-pr entry (lines 546-558)

- Add `approved_with_comments` to `outputs[0].allowed_values`
- Update verdict pattern: `"verdict\\s*=\\s*(approved|approved_with_comments|changes_requested|needs_human)"`
- Make gate tag conditional: `"(?:%%REVIEW_GATE::(LOOP_REQUIRED|CLEAR)%%|verdict\\s*=\\s*approved_with_comments)"`
- Add pattern example: `"verdict = approved_with_comments"` (no gate tag)

### 2. `merge-prs.yaml` — review-pr step `on_result` block

Add explicit `approved_with_comments` routing condition. Without this, `_check_unrouted_verdict_values` in `rules_verdict.py` will fire an ERROR when `approved_with_comments` is added to `allowed_values`. The `implementation.yaml`, `implementation-groups.yaml`, and `remediation.yaml` recipes already have this routing.

### 3. `tests/contracts/test_skill_contracts.py` — lines 181-196

Two tests assert unconditional presence of `%%REVIEW_GATE%%`:
- `test_review_gate_loop_required_pattern_in_review_pr_contracts` (line 181)
- `test_review_gate_clear_pattern_in_review_pr_contracts` (line 186)

Both must be updated to reflect the conditional gate tag pattern.

## Downstream Safety

- `review_gate_post_hook.py` — parses gate tags from result text, not from contract patterns. Unaffected: if no gate tag is present, the hook exits cleanly at line 111.
- `review_loop_gate.py` — reads `review_gate_state.json`. No state file is written for `approved_with_comments`, so the gate sees no file and exits 0. Unaffected.
- Recipe routing (`on_result` blocks) — `implementation.yaml`, `implementation-groups.yaml`, and `remediation.yaml` already route `approved_with_comments` → `resolve_review`. Only `merge-prs.yaml` needs updating.

## Blast Radius

LOW — 2 files changed (`skill_contracts.yaml`, `merge-prs.yaml`), 2 tests updated. No Python code changes. The generated recipe contracts already show the desired end state.

## Investigation Reference

investigation_path = /home/talon/projects/generic_automation_mcp/.autoskillit/temp/investigate/investigation_headless-adjudication-instability_2026-05-02_235500.md

### PR DESCRIPTION
**Pull request by Trecek**
## Summary

The contract adjudication system has a structural weakness: `expected_output_patterns` uses AND semantics (all patterns must match every session), but some patterns correspond to tokens that are conditionally emitted depending on the verdict value. When a legitimate skill output omits a conditional token, `_check_expected_patterns` returns `False`, producing a terminal `CONTRACT_VIOLATION`. This affects four skills: `review-pr`, `resolve-design-review`, `review-design`, and `diagnose-ci`. The test suite missed all four because: (1) `pattern_examples` don't cover all `allowed_values`, (2) the `pattern-examples-match` rule only checks that each pattern matches *at least one* example rather than *every* example, and (3) no test cross-validates `allowed_values` against recipe `on_result` routing conditions.

Part A fixes the four contract entries in `skill_contracts.yaml`, the `merge-prs.yaml` routing gap, and the two incorrect test assertions — with failing tests written first for each.

Part B will add three new semantic rules to `rules_contracts.py` and `rules_verdict.py` that make this entire class of bug structurally impossible.

Closes #1644

## Implementation Plan

Plan file: `/home/talon/projects/autoskillit-runs/remediation-20260502-192858-898878/.autoskillit/temp/rectify/rectify_contract_divergence_immunity_2026-05-02_192858_part_a.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code) via AutoSkillit

## Token Usage Summary

| Step | uncached | output | cache_read | cache_write | count | time |
|------|----------|--------|------------|-------------|-------|------|
| rectify | 8.1k | 18.1k | 1.1M | 87.5k | 1 | 9m 57s |
| dry_walkthrough | 79 | 22.1k | 2.4M | 121.2k | 2 | 25m 6s |
| implement | 2.5k | 53.0k | 4.4M | 163.5k | 2 | 17m 21s |
| prepare_pr | 60 | 7.0k | 204.5k | 58.9k | 1 | 2m 0s |
| compose_pr | 59 | 2.3k | 178.4k | 36.5k | 1 | 47s |
| review_pr | 843 | 65.1k | 1.7M | 346.0k | 2 | 21m 26s |
| resolve_review | 706 | 56.5k | 5.1M | 341.1k | 2 | 25m 31s |
| **Total** | 12.4k | 224.1k | 15.0M | 1.2M | | 1h 42m |
## Token Efficiency

| Step | LoC Changed | cache_read/LoC | cache_write/LoC | output/LoC |
|------|-------------|----------------|-----------------|------------|
| rectify | 0 | — | — | — |
| dry_walkthrough | 0 | — | — | — |
| implement | 717 | 6095.7 | 228.0 | 73.9 |
| assess | 29 | 76580.0 | 6696.9 | 614.1 |
| prepare_pr | 0 | — | — | — |
| compose_pr | 0 | — | — | — |
| **Total** | **746** | 13980.6 | 887.0 | 161.3 |

### CODE REVIEW

**Review by Trecek — COMMENTED** [review]
AutoSkillit PR Review — Verdict: changes_requested

File: src/autoskillit/recipe/rules_contracts.py [hunk 1]
```diff
@@ -402,3 +402,112 @@ def _check_result_field_drift(ctx: ValidationContext) -> list[RuleFinding]:
...
+        contract = get_skill_contract(name, manifest)
+        if not contract or not contract.pattern_examples:
+            continue
+
+        skill_dict = manifest.get("skills", {}).get(name, {})
+        for output in skill_dict.get("outputs", []):
+            if "allowed_values" not in output:
+                continue
+            output_name: str = output["name"]
+            for value in output["allowed_values"]:
```
Comment (Trecek) [inline src/autoskillit/recipe/rules_contracts.py:448]: [critical] bugs: Indentation bug — `for value in output["allowed_values"]:` is at the same indentation level as the outer `for output in skill_dict.get("outputs", []):` loop, not nested inside it. This means (1) if the outer loop iterates multiple outputs, only the last output's allowed_values are ever processed; (2) if `outputs` is empty the outer loop never runs, so `output` is unbound and this line raises `UnboundLocalError`. Fix: indent lines 441-456 four spaces further so the `for value` loop lives inside the `for output` body.

File: src/autoskillit/recipe/rules_verdict.py [hunk 2]
```diff
@@ -209,3 +209,74 @@ def _check_verdict_routing_asymmetry(ctx: ValidationContext) -> list[RuleFinding
...
+
+
+# Regex to extract a literal verdict value from a `when` expression.
+# Handles both template form: ${{ result.verdict }} == value
+# and bare dot-access form:    result.verdict == 'value'
+_VALUE_FROM_WHEN_RE = re.compile(
+    r"result\.(?P<output>[\w]+)"  # result.<output_name>
+    r"\s*\}?\}?"  # optional closing braces from template form
+    r"\s*==\s*"  # equality operator
+    r"'?(?P<value>\w+)'?"  # optional single-quoted value
```
Comment (Trecek) [inline src/autoskillit/recipe/rules_verdict.py]: [warning] defense: The `'?(?P<value>\w+)'?` pattern makes surrounding single-quotes independent and optional, so it matches malformed input like `'approved` (leading quote only) or `approved'` (trailing quote only). Quotes should be a matched pair — e.g., `(?:'(?P<value>\w+)'|(?P<value>\w+))` — or validate/strip after extraction.

File: src/autoskillit/recipe/rules_verdict.py [hunk 3]
```diff
@@ -209,3 +209,74 @@ def _check_verdict_routing_asymmetry(ctx: ValidationContext) -> list[RuleFinding
             )
 
     return findings
+
+
+# Regex to extract a literal verdict value from a `when` expression.
+# Handles both template form: ${{ result.verdict }} == value
+# and bare dot-access form:    result.verdict == 'value'
+_VALUE_FROM_WHEN_RE = re.compile(
```
Comment (Trecek) [inline src/autoskillit/recipe/rules_verdict.py:217]: [warning] cohesion: The comment says this regex handles both template form (`${{ result.verdict }} == value`) and bare dot-access, but `\}?\}?` only optionally strips the closing `}}` — the opening `${{` is never consumed. The regex works via substring search, but the comment implies structural template handling that is not implemented. Correct the comment or make the template anchor explicit.

File: tests/contracts/test_skill_contracts.py [hunk 4]
```diff
@@ -177,20 +177,155 @@ def test_skill_contracts_yaml_open_research_pr_removed(skills):
...
+        (o for o in contracts["skills"]["review-pr"].get("outputs", []) if o["name"] == "verdict"),
+        None,
+    )
+    assert verdict_output is not None
+    allowed = set(verdict_output.get("allowed_values", []))
+
+    # Match only lowercase verdict values (review-pr convention: approved, changes_requested…).
+    # Excludes all-uppercase review-design verdicts (GO, REVISE, STOP) which appear in the
+    # same recipe files under different steps.
+    verdict_route_re = re.compile(r"result\.verdict\s*}}\s*==\s*([a-z][a-z_]*)")
```
Comment (Trecek) [inline tests/contracts/test_skill_contracts.py:271]: [warning] tests: `verdict_route_re = re.compile(r"result\.verdict\s*}}\s*==\s*([a-z][a-z_]*)")` — the character class `[a-z_]` excludes digits, so a future verdict value containing a digit (e.g., `approved_v2`) would be silently missed. Change to `[a-z][a-z0-9_]*`.

File: tests/hooks/test_review_gate_post_hook.py [hunk 5]
```diff
@@ -274,3 +274,39 @@ def test_pr_number_extracted_from_run_python_args(tmp_path):
...
+        "check_review_loop_called": False,
+        "pr_number": "1290",
+        "set_at": "2026-04-26T04:30:00+00:00",
+    }
+    state_path.write_text(json.dumps(existing))
+
+    _run_hook(event, tmp_dir=tmp_path)
+    state = _read_state(tmp_path)
+    assert state is not None, (
+        "approved_with_comments must not clear an existing LOOP_REQUIRED state — "
```
Comment (Trecek) [inline tests/hooks/test_review_gate_post_hook.py:309]: [warning] tests: Part 2 only asserts `state["gate"] == "LOOP_REQUIRED"` but does not verify the full state is unchanged. The hook could silently mutate `review_verdict`, `pr_number`, or `set_at` without this test catching it. Assert `state == existing` (the full dict) to prevent silent field mutations.

File: tests/recipe/test_rules_contracts.py [hunk 6]
```diff
@@ -412,3 +412,190 @@ def test_planner_recipe_has_no_result_field_drift(self) -> None:
...
+        "(defer to missing-pattern-examples rule)"
+    )
+
+
+# ---------------------------------------------------------------------------
+# Integration test: all bundled recipes pass the new contract immunity rules
+# ---------------------------------------------------------------------------
+
+
+def test_bundled_recipes_pass_all_new_contract_rules() -> None:
```
Comment (Trecek) [inline tests/recipe/test_rules_contracts.py:579]: [warning] tests: `recipe_files = sorted(recipes_dir.glob("*.yaml"))` — the test will pass vacuously if `recipes_dir` does not exist or contains no YAML files. Add `assert recipe_files, f"No bundled recipes found at {recipes_dir}"` before the loop to prevent a silent false-green on partial installs.

File: tests/recipe/test_rules_contracts.py [hunk 7]
```diff
@@ -412,3 +412,190 @@ def test_planner_recipe_has_no_result_field_drift(self) -> None:
...
+    monkeypatch.setattr(_rc, "load_bundled_manifest", lambda: manifest)
+    recipe = _make_recipe_with_skill("/autoskillit:test-skill")
+    findings = run_semantic_rules(recipe)
+    rule_findings = [f for f in findings if f.rule == "all-examples-match-all-patterns"]
+    assert len(rule_findings) >= 1, (
+        "all-examples-match-all-patterns must fire when 'verdict = stop' example "
+        "does not match the '%%TAG%%' pattern"
+    )
+    assert rule_findings[0].severity == Severity.ERROR
+    assert "%%TAG%%" in rule_findings[0].message or "verdict = stop" in rule_findings[0].message
```
Comment (Trecek) [inline tests/recipe/test_rules_contracts.py:516]: [warning] tests: `assert "%%TAG%%" in rule_findings[0].message or "verdict = stop" in rule_findings[0].message` — the OR means either half can be absent without failing. If both tokens are expected, use `and`; if only one is required, remove the other branch and add a comment explaining why.

**Review by Trecek — COMMENTED** [review]
AutoSkillit review: 1 critical blocking finding detected. See inline comments for details.

### ⚠️ Outside Diff Range

These findings target lines not in the diff and could not be posted as inline comments:

**src/autoskillit/recipe/rules_verdict.py**
- **L25** [warning/cohesion]: `_get_allowed_values_for_skill` re-implements the same raw manifest walk already done in rules_contracts.py; two scattered implementations of the same manifest-to-allowed-values extraction.
- **L42** [warning/cohesion]: `_extract_skill_name` in rules_verdict.py duplicates `resolve_skill_name` from rules_contracts.py with a different name and subtly different semantics (match vs search).
- **L44** [warning/cohesion]: `_extract_skill_name` uses `re.match` while `resolve_skill_name` uses `re.search`; match anchors to string start so a skill_command with any prefix would be silently dropped.

File: src/autoskillit/recipe/rules_contracts.py (hunk 1 above)
Comment (Trecek) [reply src/autoskillit/recipe/rules_contracts.py:448]: Investigated — this is intentional. The `for value in output["allowed_values"]:` at line 441 is at 12-space indentation (3 levels), while `for output in skill_dict.get("outputs", []):` at line 437 is at 8-space indentation (2 levels). These are different indentation levels — the inner loop is correctly nested. The git diff in commit cb3169ad confirms identical structure.

File: src/autoskillit/recipe/rules_verdict.py (hunk 2 above)
Comment (Trecek) [reply src/autoskillit/recipe/rules_verdict.py]: Agreed — fixed in 31c424df7f61d433a29c5304565ebad05110b5d4. Changed `'?(?P<value>\w+)'?` to `(?P<value>'\w+'|\w+)` so the pattern requires balanced quotes (both or neither). Added `.strip("'")` at the call site to extract the bare word from matched quoted values.

File: src/autoskillit/recipe/rules_verdict.py (hunk 3 above)
Comment (Trecek) [reply src/autoskillit/recipe/rules_verdict.py:217]: Investigated — this is intentional. The regex is used with `finditer` (substring search), not `fullmatch`. For the template form `${{ result.verdict }} == value`, `finditer` locates `result.verdict` mid-string and `\}?\}?` optionally consumes the closing `}}`. The opening `${{{` need not be consumed — partial matching is the correct and intentional design. The comment accurately describes both handled forms.

File: tests/contracts/test_skill_contracts.py (hunk 4 above)
Comment (Trecek) [reply tests/contracts/test_skill_contracts.py:271]: Valid observation — flagged for design decision. The comment at lines 268-270 explicitly documents `[a-z_]` as intentional to exclude uppercase review-design verdicts (GO, REVISE, STOP). Whether future verdict names may contain digits (e.g. `approved_v2`) is a naming-convention decision. Extending to `[a-z0-9_]` is safe but requires confirming the digit-exclusion is not load-bearing for other validators.

File: tests/hooks/test_review_gate_post_hook.py (hunk 5 above)
Comment (Trecek) [reply tests/hooks/test_review_gate_post_hook.py:309]: Agreed — fixed in 496aec93a01ff513ddd2013ae6ce3055532267e9. Replaced single-field `state["gate"] == "LOOP_REQUIRED"` check with `state == existing` full dict equality, so any silent mutation to `review_verdict`, `pr_number`, `set_at`, or `check_review_loop_called` will now fail the test.

File: tests/recipe/test_rules_contracts.py (hunk 6 above)
Comment (Trecek) [reply tests/recipe/test_rules_contracts.py:579]: Investigated — this is intentional. The guard requested (`assert recipe_files`) is already present at line 588: `assert recipe_files, "No bundled recipes found"`. The vacuous-pass scenario is already covered; this appears to be a stale comment reviewing a pre-fix version of the code.

File: tests/recipe/test_rules_contracts.py (hunk 7 above)
Comment (Trecek) [reply tests/recipe/test_rules_contracts.py:516]: Investigated — this is intentional. The message template is `f"Skill '{name}': example {preview} does not match pattern {pattern!r}."`. For the test's specific failure case ('verdict = stop' failing pattern '%%TAG%%'), `preview` = `repr('verdict = stop')` and `pattern!r` = `'%%TAG%%'` — both tokens are always simultaneously present in the message. The OR is not a weakness; both branches are always satisfied for this failure scenario.

File: tests/contracts/test_skill_contracts.py [hunk 8]
```diff
@@ -177,20 +177,155 @@ def test_skill_contracts_yaml_open_research_pr_removed(skills):
...
+
+    Scans implementation.yaml, remediation.yaml, implementation-groups.yaml, and
+    merge-prs.yaml for result.verdict routing conditions. Any value routed in a recipe
+    but absent from skill_contracts.yaml allowed_values will trigger the
+    unrouted-verdict-value semantic rule at recipe-load time.
+    """
+    recipes_dir = Path(__file__).parents[2] / "src/autoskillit/recipes"
+    target_files = [
+        "implementation.yaml",
+        "remediation.yaml",
```
Comment (Trecek) [inline tests/contracts/test_skill_contracts.py:256]: [warning] tests: Duplicate `assert verdict_output is not None` at lines 255 and 256. The second assertion is redundant and adds noise without protection.

File: tests/contracts/test_skill_contracts.py [hunk 9]
```diff
@@ -177,20 +177,155 @@ def test_skill_contracts_yaml_open_research_pr_removed(skills):
...
+    """
+    assert "review-pr" in skills
+    verdict_output = next(
+        (o for o in skills["review-pr"].get("outputs", []) if o["name"] == "verdict"),
+        None,
+    )
+    assert verdict_output is not None
+    allowed = verdict_output.get("allowed_values", [])
+    examples = skills["review-pr"].get("pattern_examples", [])
+    missing = [v for v in allowed if not any(f"verdict = {v}" in ex for ex in examples)]
```
Comment (Trecek) [inline tests/contracts/test_skill_contracts.py:211]: [warning] tests: The check `f"verdict = {v}" in ex` is a plain substring match rather than a regex search. This is inconsistent with test_every_pattern_example_satisfies_all_patterns (L230) which uses re.search. Consider `re.search(f'verdict\\s*=\\s*{re.escape(v)}', ex)` for consistency with the pattern-matching contract semantics.

File: tests/contracts/test_skill_contracts.py [hunk 10]
```diff
@@ -177,20 +177,155 @@ def test_skill_contracts_yaml_open_research_pr_removed(skills):
...
+    unrouted-verdict-value semantic rule at recipe-load time.
+    """
+    recipes_dir = Path(__file__).parents[2] / "src/autoskillit/recipes"
+    target_files = [
+        "implementation.yaml",
+        "remediation.yaml",
+        "implementation-groups.yaml",
+        "merge-prs.yaml",
+    ]
+    contracts = yaml.safe_load(_CONTRACTS_YAML.read_text())
```
Comment (Trecek) [inline tests/contracts/test_skill_contracts.py:260]: [warning] tests: The regex `r"result\.verdict\s*}}\s*==\s*([a-z][a-z_]*)"` will silently miss quoted verdict values — if a recipe writes `== 'approved_with_comments'` with single quotes, the match fails and routed_values is empty, giving a vacuous pass. The character class `[a-z_]` also excludes digits from future value names.

File: src/autoskillit/recipe/rules_contracts.py [hunk 11]
```diff
@@ -402,3 +402,112 @@ def _check_result_field_drift(ctx: ValidationContext) -> list[RuleFinding]:
...
+)
+def _check_example_covers_all_allowed_values(ctx: ValidationContext) -> list[RuleFinding]:
+    """Error when any allowed_value has no corresponding pattern_examples entry.
+
+    Reads allowed_values from the raw manifest dict (not SkillContract, which does not
+    promote allowed_values into SkillOutput). An example 'covers' a value when the
+    example text contains a match for '{output_name}\\s*=\\s*{re.escape(value)}'.
+    """
+    findings: list[RuleFinding] = []
+    manifest = load_bundled_manifest()
```
Comment (Trecek) [inline src/autoskillit/recipe/rules_contracts.py]: [warning] defense: `load_bundled_manifest()` is called without try/except. A manifest I/O failure will propagate as an unhandled exception and abort the entire validation pass, unlike the analogous `_get_allowed_values_for_skill` in rules_verdict.py which wraps the call defensively.

File: src/autoskillit/recipe/rules_contracts.py [hunk 12]
```diff
@@ -402,3 +402,112 @@ def _check_result_field_drift(ctx: ValidationContext) -> list[RuleFinding]:
...
+
+    for step_name, step in ctx.recipe.steps.items():
+        if step.tool != "run_skill":
+            continue
+        skill_cmd = step.with_args.get("skill_command", "")
```
Comment (Trecek) [inline src/autoskillit/recipe/rules_contracts.py]: [warning] defense: `step.with_args.get(...)` is unguarded. If `with_args` is None, this raises AttributeError. The analogous call in rules_verdict.py defensively uses `(step.with_args or {}).get(...)`. Fix both this and the same pattern at L483.

File: src/autoskillit/recipe/rules_contracts.py [hunk 13]
```diff
@@ -402,3 +402,112 @@ def _check_result_field_drift(ctx: ValidationContext) -> list[RuleFinding]:
...
+)
+def _check_all_examples_match_all_patterns(ctx: ValidationContext) -> list[RuleFinding]:
+    """Error when any pattern_examples entry fails any expected_output_pattern.
+
+    The existing pattern-examples-match rule checks ∀ pattern, ∃ example (pattern not dead).
+    This rule checks ∀ example, ∀ pattern (pattern not conditional). Together they form a
+    complete constraint: patterns are both necessary and sufficient for all output paths.
+    """
+    findings: list[RuleFinding] = []
+    manifest = load_bundled_manifest()
```
Comment (Trecek) [inline src/autoskillit/recipe/rules_contracts.py]: [warning] defense: `load_bundled_manifest()` at L478 is called without try/except in `_check_all_examples_match_all_patterns`, same issue as L423. Manifest I/O failures propagate uncaught.

File: src/autoskillit/recipe/rules_contracts.py [hunk 14]
```diff
@@ -402,3 +402,112 @@ def _check_result_field_drift(ctx: ValidationContext) -> list[RuleFinding]:
...
+
+    for step_name, step in ctx.recipe.steps.items():
+        if step.tool != "run_skill":
+            continue
+        skill_cmd = step.with_args.get("skill_command", "")
```
Comment (Trecek) [inline src/autoskillit/recipe/rules_contracts.py]: [warning] defense: `step.with_args.get(...)` is unguarded at L483, same issue as L428 — should be `(step.with_args or {}).get(...)`.

**Review by Trecek — COMMENTED** [review]
AutoSkillit PR Review — Verdict: approved_with_comments

**Review by Trecek — COMMENTED** [review]
AutoSkillit review: warning-only findings detected. See inline comments — no blocking changes required.

### ⚠️ Outside Diff Range

These findings target lines not in the diff and could not be posted as inline comments:

**src/autoskillit/recipe/rules_verdict.py**
- **L42** [warning/cohesion]: `_extract_skill_name` in rules_verdict.py duplicates `resolve_skill_name` from contracts.py (already used in rules_contracts.py) with slightly different matching semantics (.match vs .search). Import and reuse the shared helper instead.
- **L25** [warning/cohesion]: `_get_allowed_values_for_skill` re-implements manifest lookup that could be composed from the shared `get_skill_contract` helper (used in rules_contracts.py). Two callers now maintain separate manifest-traversal paths for the same allowed_values data.
- **L22** [info/cohesion]: `_SKILL_NAME_RE` is redefined locally in rules_verdict.py; the same pattern already exists as a shared constant in contracts.py. Import the constant or reuse resolve_skill_name to remove this duplication.

File: tests/contracts/test_skill_contracts.py (hunk 8 above)
Comment (Trecek) [reply tests/contracts/test_skill_contracts.py:256]: Investigated — this is intentional. The two `assert verdict_output is not None` assertions appear in different independent test functions (`test_review_pr_verdict_allowed_values_includes_approved_with_comments` L191 and `test_review_pr_pattern_examples_cover_all_verdicts` L208). Each test function must carry its own precondition guard — relying on another test's assertion would violate test isolation; the two assertions are not duplicates.

File: tests/contracts/test_skill_contracts.py (hunk 9 above)
Comment (Trecek) [reply tests/contracts/test_skill_contracts.py:211]: Investigated — this is intentional. `f"verdict = {v}" in ex` is an intentionally strict substring match for the canonical skill output token format `verdict = {v}` (with spaces). Using `re.search` with `\s*` would be incorrectly permissive (would accept `verdict=value` without spaces). The `re.search` in `test_every_pattern_example_satisfies_all_patterns` applies regex PATTERN strings from `expected_output_patterns` — a fundamentally different operation from checking a specific value literal.

File: tests/contracts/test_skill_contracts.py (hunk 10 above)
Comment (Trecek) [reply tests/contracts/test_skill_contracts.py:260]: Investigated — this is intentional. Verified all 4 target recipe files: review-pr verdict routing uses UNQUOTED values in every case (`== changes_requested`, `== approved_with_comments`, `== needs_human`). The quoted format (`== 'real_fix'`) is exclusively for resolve-review verdicts, which are filtered out by the `[a-z][a-z_]*` capture group. The test is not vacuous — it correctly captures all three review-pr routing values.

File: src/autoskillit/recipe/rules_contracts.py (hunk 11 above)
Comment (Trecek) [reply src/autoskillit/recipe/rules_contracts.py]: Agreed — fixed in 899f8853. Added try/except around `load_bundled_manifest()` in `_check_example_covers_all_allowed_values`, matching the defensive pattern in `_get_allowed_values_for_skill` (rules_verdict.py L27-33). Returns empty findings and logs a warning on manifest I/O failure.

File: src/autoskillit/recipe/rules_contracts.py (hunk 12 above)
Comment (Trecek) [reply src/autoskillit/recipe/rules_contracts.py]: Agreed — fixed in 899f8853. Changed `step.with_args.get()` to `(step.with_args or {}).get()` at L428, matching the defensive pattern used in rules_verdict.py at lines 81, 168, and 245.

File: src/autoskillit/recipe/rules_contracts.py (hunk 13 above)
Comment (Trecek) [reply src/autoskillit/recipe/rules_contracts.py]: Agreed — fixed in 899f8853. Added try/except around `load_bundled_manifest()` in `_check_all_examples_match_all_patterns`, matching the defensive pattern in `_get_allowed_values_for_skill` (rules_verdict.py L27-33). Returns empty findings and logs a warning on manifest I/O failure.

File: src/autoskillit/recipe/rules_contracts.py (hunk 14 above)
Comment (Trecek) [reply src/autoskillit/recipe/rules_contracts.py]: Agreed — fixed in 899f8853. Changed `step.with_args.get()` to `(step.with_args or {}).get()` at L483, matching the defensive pattern used in rules_verdict.py at lines 81, 168, and 245.
